mathics/core/expression.py

```diff
@@ -199,9 +199,9 @@ def __init__(
     ):
         self.options = None
         self.pattern_sequence = False
-        if isinstance(head, str):
-            # We should fix or convert to to_expression all nonSymbol uses.
-            head = Symbol(head)
+        assert isinstance(head, BaseElement)
```

Comment:
Now that this is the case, we should change the signature in line 197 or so to:

`self, head: BaseElement, ...`

Reply:
Yep. It seems we gain 25% of the time in the doctests avoiding these asserts...